lib/IRGen/GenCall.cpp

```diff
@@ -2187,7 +2191,12 @@ class SyncCallEmission final : public CallEmission {
       break;
 
     case SILFunctionTypeRepresentation::Block:
-      adjusted.add(getCallee().getBlockObject());
+    case SILFunctionTypeRepresentation::CXXMethod:
+      if (getCallee().getRepresentation() ==
```

Comment:
Why are you handling the block and cxxmethod in same case if you need to do a different call for block vs cxx method? I think it would be better to have separate cases.

Reply:
Aha. Turns out there was actually a reason for this: we want the fall-through here for both cases. Without this weird if-inside-the-case pattern we don't `externalizeArguments` for `::Block`.